File: 2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int count = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (s[i] == '1') count++;
        }
        string ans = "";
        for (int i = 0; i < count - 1 ; i++) {
            ans += "1";
        }
        for (int i = 0; i < n - count; i++) {
            ans += "0";
        }
        ans += "1";
        return ans;
    }
};

File: LeetCode/2864-maximum-odd-binary-number/2864-maximum-odd-binary-number.java
class Solution {
    public String maximumOddBinaryNumber(String s) {
        int n = s.length();
        StringBuilder result = new StringBuilder(String.join("", Collections.nCopies(n, "0")));
        int count_1 = 0;
        for (char ch : s.toCharArray()) {
            if (ch == '1') {
                count_1++;
            }
        }
        int j = 0;
        while (count_1 > 1) {
            result.setCharAt(j, '1');
            j++;
            count_1--;
        }
        result.setCharAt(n - 1, '1');
        return result.toString();
    }
}

// class Solution {
//     public String maximumOddBinaryNumber(String s) {
//         int n = s.length();
//         StringBuilder result = new StringBuilder();
//         for (int i = 0; i < n; i++) {
//             result.append('0');
//         }
//         int i = 0;
//         for (char ch : s.toCharArray()) {
//             if (ch == '1') {
//                 if (result.charAt(n-1) == '1') {
//                     result.setCharAt(i, '1');
//                     i++;
//                 } else {
//                     result.setCharAt(n - 1, '1');
//                 }
//             }
//         }
//         return result.toString();
//     }
// }